cranelift/codegen/src/isa/zkasm/inst/emit.rs

```diff
@@ -660,11 +664,15 @@ impl MachInstEmit for Inst {
                         );
                     }
                     AMode::FPOffset(..) => {
+                        // We don't have a frame pointer on ZKASM, but it can be easily computed
+                        // based on the nominal sp.
+                        let offset = state.nominal_sp_to_fp + offset;
```

Comment:
I would perhaps move this computation to the function that calculates `offset`, and at this point the correct thing to do would be to also use `AMode::get_base_register` instead of hardcoding registers in every location where `AMode` is getting handled.

Reply:
Great idea, the same came to my mind after your comment on https://github.com/near/wasmtime/pull/193. Will do.